### ISSUE TITLE
Facilitate checking the input structure details (and raw input files)

### ISSUE THREAD

**Issue by giovannipizzi** [aiidalab/aiidalab-qe#352]
**Is your feature request related to a problem? Please describe.**
I run a calculation without relaxation. I want now to check the input structure (in particular, the lattice parameter and the input cell. However:
- in the output, the final geometry is N/A since there was no relax
- in the tree view there are only outputs but no inputs
- if I go to step 1, I can see a number of structure but 1) it's not obvious if I'm looking at the input of this calculation (I can actually change it, I run SnTe but I can swap to e.g. silicon), and 2) I only see the visualizer an not coordinates and cell in tabular form.

Analogously, there is (at least after running) no way in the tree view to see the actual SCF calculation, nor to see the raw input and output files (see screenshot):
![Screenshot from 2023-02-15 12-05-52](https://user-images.githubusercontent.com/16132028/219010775-11170417-5105-4a15-9ba5-e2dac43b28a4.png)

**Describe the solution you'd like**
It should be easy to
- visualize the input structure. I suggest to add, in the output tabs, a tab "Initial geometry" (before Final geometry) that shows the geometry: 3D view, but also cell and coordinates (possibly also downloadable in a few common formats: CIF, XSF.
- (optional) maybe show also the input structure in the tree view? Not sure how to implement this in a way that is not confusing, though
- make it obvious it's exactly the structure used in the selected workflow (e.g., disable clicking back on Step 1/2/3): now even if I relesect the workflow (but did not refresh the page) I still see the (incorrect) silicon structure from my DB, while the WF was run on SnTe.
- make it possible (a new tab, or in some obvious way from the tree view) to inspect raw inputs and outputs of all QE runs (SCF, NSCF, ...): if an experienced person wants to see them, they should be easily be able to do it (at least the raw I/O of QE, as a priority over the AiiDA parsed information)

**Describe alternatives you've considered**
The only way I see is to open a terminal now, load the workflow by PK, load the input structure, and print the coordinates and cell from the cmdline (similarly, check the `process status` of the workflow and do a `calcjob inputcat/outputcat` for each QE calc), but we of course need to avoid that people go to the terminal for common tasks.

**Comment by giovannipizzi** [comment]
PS - I now see that probably I could see the inputs by navigating deep into the PwBandsWorkChain. 
I cannot provide more feedback because of the crash described in #353, but I will open a new issue #354 to describe some improvements

**Comment by unkcpz** [comment]
> I now see that probably I could see the inputs by navigating deep into the PwBandsWorkChain.

Sadly, you can't. Even we use the generic WorkChain viewer, the inputs are not accessible, since the inputs not defined and show for it. But sure, this is very important feature and I'll see how to add it.

**Comment by danielhollas** [comment]
> in the tree view there are only outputs but no inputs

I agree that this would be a useful feature to be able to inspect input nodes. We could add the corresponding `inputs` directory with all the input nodes. I've opened the corresponding issue on AWB https://github.com/aiidalab/aiidalab-widgets-base/issues/432
A possible disadvantage is performence, I noticed that for large workchains the tree is loading very slowly so this might add extra overhead since all those nodes would need to be fetched from DB.

This would allow us to inspect input geometries and other parameters. However, it wouldn't allow to inspect the QE input since that is generated by the CalcJob and stored in the remote_folder and not fetched back.

**Comment by giovannipizzi** [comment]
Thanks! Two comments.

Performance-wise, maybe it's possible to just load the name, label, and PK/UUID of the various nodes, and get their contents only if/when they are clicked?

For the raw QE input - it's not correct that the raw input is not stored in AiiDA. It is stored as files directly inside the CalcJob node itself. Therefore, one could either think to add a special "raw input" leaf under "inputs", that actually shows the raw files stored in the CalcJob node, or even have a different directory "Raw inputs" parallel to "inputs". I'm not sure if this is more or less intuitive for a (non-AiiDA) user. If we put a directory "Raw inputs", maybe we should name the other inputs as something else, e.g. "AiiDA inputs" or "node inputs" or some other name?

**Comment by unkcpz** [comment]
I think we have all ingredients widgets for showing everything related to process, implemented in aiidalab-widgets-base already. The [`process_list.ipynb`](https://github.com/aiidalab/aiidalab-widgets-base/blob/master/notebooks/process_list.ipynb)  of AWB have all such followers. I'll take a look at this and see how far we can reach with the current widgets.

**Comment by unkcpz** [comment]
Working on adding test for node tree widgets, some thoughts about how to improve the performance of the widget. As mentioned by @danielhollas, for a large workchain with lots of nested process, the widget when initialize it will traverse all the nested nodes to the end leaf nodes.  
Every tree node has a bool attribute named `opened`, we can only traverse for `opened=true` nodes. For `opened=False`nodes, it at most unfolded two more levels.
When looking at this code, I make some mistakes to mess up the name for variables between the node of tree widget and node of aiida, the variable name needs to be clean up to avoid ambiguity.

**Comment by unkcpz** [comment]
We didn't manage to address this issue for the up-coming release, so I'll move it to the next milestone.
